demo/visualizer.js

```diff
@@ -244,12 +241,30 @@ shakaDemo.Visualizer = class {
   }
 
   /** @private */
-  ageUpdates_() {
-    for (const update of this.updates_) {
-      update.age += shakaDemo.Visualizer.updateFrequency_;
-    }
+  pruneUpdates_() {
+    const allBufferedInfo = this.player_.getBufferedInfo();
```

Comment:
Would this work with Safari's native HLS?

Reply:
Changed to fallback to allBufferedInfo.total when the individual arrays aren't present.